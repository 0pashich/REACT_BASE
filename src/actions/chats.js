import firebase from 'firebase'
export const ADD_CHAT = 'CHATS::ADD_CHAT'
export const REMOVE_CHAT = 'CHATS::REMOVE_CHAT'

export const addChat = (chatId, name) => ({
    type: ADD_CHAT,
    payload: {
        id: chatId,
        name,
    },
})

export const removeChat = (chatId) => ({
    type: REMOVE_CHAT,
    payload: {
        chatId,
    },
})

export const addChatWithFirebase = (chatId, name) => {
    return () => {
        firebase.database().ref('chats').child(chatId).push({ id: chatId, name: name, })
        console.log('addChatWithFirebase')
    }
}
export const deleteChatWithFirebase = (chatId) => async () => {
    firebase.database().ref("chats").child(chatId).remove();
    console.log('deleteChatWithFirebase')
};

export const subscribeOnChatsChangings = () => {
    return (dispatch, getState) => {
        firebase
            .database()
            .ref('chats')
            .on('child_added', (snapshot) => {
                console.log('child_added', snapshot.val())
                //  snapshot.val()[])
                // dispatch(addChat( snapshot.val()))
            })

        firebase
            .database()
            .ref('messages')
            .on('child_changed', (snapshot) => {
                console.log('child_changed', snapshot.val())

                // dispatch(addChat(chatId, snapshot.val()))
            })
    }
}



// export const sendMessageToBot = (chatId, message) => {
//     return () => {
//         firebase.database().ref('messages').child(chatId).push(message)

//         let timer = setTimeout(() => {
//             firebase
//                 .database()
//                 .ref('messages')
//                 .child(chatId)
//                 .push({
//                     id: `message${Date.now()}`,
//                     author: AUTHORS.BOT,
//                     text: 'Привет, я - бот!',
//                 })

//             clearTimeout(timer)
//         }, 1500)
//     }
// }

// export const subscribeOnChatChangings = (chatId) => {
//     return (dispatch, getState) => {
//         firebase
//             .database()
//             .ref('messages')
//             .child(chatId)
//             .on('child_added', (snapshot) => {
//                 console.log('child_added', snapshot.val())

//                 dispatch(addMessage(chatId, snapshot.val()))
//             })

//         firebase
//             .database()
//             .ref('messages')
//             .child(chatId)
//             .on('child_changed', (snapshot) => {
//                 console.log('child_changed', snapshot.val())

//                 dispatch(addMessage(chatId, snapshot.val()))
//             })
//     }
// }