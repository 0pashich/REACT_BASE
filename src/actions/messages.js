import { AUTHORS } from '../components/App/constants'
export const ADD_MESSAGE = 'MESSAGES:ADD_MESSAGE'


export const addMessage = (chatId, message) => ({
    type: ADD_MESSAGE,
    payload: {
        chatId,
        message,
    },
})

export const addMessageWithThunk = (chatId, message) => (dispatch, getState) => {
    dispatch(addMessage(chatId, message));
    if (message.author !== AUTHORS.BOT) {
        const botMessage = { id: `botmessage${Date.now()}`, text: 'Привет!', author: AUTHORS.BOT };
        setTimeout(() => dispatch(addMessage(chatId, botMessage)), 2000);
    }
}
