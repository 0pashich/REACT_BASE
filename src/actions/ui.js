export const SET_CURENT_CHAT = 'UI::SET_CURENT_CHAT'
export const SET_DRAWER_OPEN = 'UI::SET_DRAWER_OPEN'

export const setCurentChat = (curentChat) => ({
    type: SET_CURENT_CHAT,
    payload: {
        curentChat,
    },
})

export const setDrawerOpen = (drawerOpen) => ({
    type: SET_DRAWER_OPEN,
    payload: {
        drawerOpen,
    },
})


