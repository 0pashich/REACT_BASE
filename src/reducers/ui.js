import { SET_CURRENT_CHAT, SET_DRAWER_OPEN } from '../actions/ui'


const initialState = {
    drawerOpen: false,
    currentChat: {},
}

export default function uiReducer(state = initialState, action) {
    switch (action.type) {
        case SET_CURRENT_CHAT: {
            return {
                ...state,
                currentChat: action.payload.currentChat,
            }
        }
        case SET_DRAWER_OPEN: {
            return {

                ...state,
                drawerOpen: action.payload.drawerOpen,
            }
        }
        default:
            return state
    }
}