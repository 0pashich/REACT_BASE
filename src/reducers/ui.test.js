import { render, screen } from "@testing-library/react";
import uiReducer from "./ui";
import { setCurrentChat } from '../actions/ui'

describe("UI", () => {


    it('returns correct state after SET_CURRENT_CHAT action', () => {
        const initialState = {
            chat1: {
                id: 'chat1',
                name: 'Чат 1',
            },
        }
        const uiStore = uiReducer(initialState, setCurrentChat(initialState));
        expect(uiStore).toMatchSnapshot();
    });


    // it('matches snapshot online', () => {
    //     const component = render(<Message author="Anton" text="Hello" />)

    //     expect(component).toMatchSnapshot()
    // })

    // it("should contain message text 'Hello'", () => {
    //     render(<Message author="Anton" text="Hello" />)

    //     const wrapper = screen.getByText(/Hello/i)

    //     expect(wrapper).toBeInTheDocument()
})
