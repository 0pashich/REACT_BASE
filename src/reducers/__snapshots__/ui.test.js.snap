// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI returns correct state after SET_CURRENT_CHAT action 1`] = `
Object {
  "chat1": Object {
    "id": "chat1",
    "name": "Чат 1",
  },
  "currentChat": Object {
    "chat1": Object {
      "id": "chat1",
      "name": "Чат 1",
    },
  },
}
`;
